src/content/en/tools/workbox/modules/workbox-webpack-plugin.md

```diff
@@ -0,0 +1,82 @@
+project_path: /web/tools/workbox/_project.yaml
+book_path: /web/tools/workbox/_book.yaml
+description: The module guide for workbox-webpack-plugin.
+
+{# wf_updated_on: 2017-12-15 #}
+{# wf_published_on: 2017-12-15 #}
+{# wf_blink_components: Blink>ServiceWorker #}
+
+# Workbox webpack Plugin  {: .page-title }
+
+## What's the Workbox webpack Plugin?
+
+This is a plugin for the [webpack](https://webpack.js.org/) which generates a list of URLs to
+[precache](workbox-precaching) (known as a "precache manifest") based on the assets in a webpack
+compilation.
+
+## Considerations
+
+There are a few decisions to make which will determine the right configuration for your use case.
+
+### Should the plugin create your service worker file?
+
+If you leave out the `swSrc` configuration option, the plugin will create a service worker file for
+you and automatically add it to the webpack asset pipeline. This is referred to as `generateSW`
+mode.
+
+{% include "web/tools/workbox/_shared/when-to-use-generate-sw.md" %}
+
+If you set `swSrc` to the path of an existing service worker file (which does not have to be
+otherwise configured as a webpack asset) then the plugin will generate a list of URLs to precache
+and add a reference to that precache manifest, along with a reference to the Workbox runtime code,
+to your file. It will otherwise leave the file as-is. This is referred to as `injectManifest` mode.
+
+{% include "web/tools/workbox/_shared/when-to-use-inject-manifest.md" %}
+
+### Do you need to cache additional, non-webpack assets?
+
+By default, the plugin will generate a precache manifest that contains URLs for all of the assets
+that come from the current webpack compilation. Any assets that webpack doesn't "know" about will
+not be picked up.
+
+If you need to precache additional assets that are managed outside of webpack, then you can
+use the `globDirectory` and `globPatterns` options to specify how to find those additional assets.
+
+If you decide to use `globDirectory` and `globPatterns`, the following applies:
+
+- The [`glob` pattern matching](https://github.com/isaacs/node-glob#glob-primer) will be performed
+against the local file system, and the directory that `globDirectory` is set to must exist at the
+time the plugin runs. This might run afoul of
+[`webpack-dev-server` configurations](https://github.com/webpack/webpack-dev-server), where an
+in-memory file system is used.
+- The options that work with `glob`-based precache manifests, like `manifestTranforms` and
+`modifyUrlPrefix`, can also be used, but they'll apply **only** to the entries that are matched via
+`glob` patterns, and not to any assets that are picked up via the webpack compilation.
+
```

Comment:
It's great that this information is here.   However it would be good if the workbox docs for `globDirectory` and `globPatterns` explicitly state that they are an "escape hatch" rather than something that you should need to configure.  Left to my own devices I had assumed there was a default `globPattern` of `'**/*.*` or something.  Not so - but I didn't realise.

Reply:
There's currently the following disclaimer:

> This only needs to be set if you want to precache local files matching <code>glob</code> patterns. It does not need to be set if you have <code>self.__precacheManifest</code> defined via a file included in <code>importScripts</code>.

But I'll make it clearer that developers who are using the webpack plugin are the ones who wouldn't need to set them.